### PR TITLE
Allow Spawn to be unsized

### PR DESCRIPTION
**Pull request by cramertj**
This would allow me to save an allocation in a very performance-sensitive area.

### CODE REVIEW

File: src/task_impl/mod.rs [hunk 1]
```diff
@@ -242,15 +242,17 @@ impl<T> Spawn<T> {
     /// Get a mutable reference to the object the Spawn is wrapping.
     pub fn get_mut(&mut self) -> &mut T {
         &mut self.obj
-    }
+    } 
+}
 
+impl<T> Spawn<T> {
```
Comment (alexcrichton) [inline src/task_impl/mod.rs]: Mind leaving these in the same impl block, but adding `where Self: Sized` to this method?

Comment (alexcrichton) [reply src/task_impl/mod.rs]: If you're super enterprising, you could collapse all the below impl blocks into one with the same strategy as well :)

(no need to happen in this PR though)

File: tests/all.rs [hunk 2]
```diff
@@ -359,3 +359,17 @@ fn option() {
...
 }
+
+#[test]
+fn spawn_does_unsize() {
+    #[derive(Clone, Copy)]
+    struct EmptyNotify;
+    impl executor::Notify for EmptyNotify {
+        fn notify(&self, _: usize) { panic!("Cannot notify"); }
+    }
+    static EMPTY: &EmptyNotify = &EmptyNotify;
```
Comment (alexcrichton) [inline tests/all.rs]: Mind throwing a `'static` in here to fix the compile on 1.15?

**Comment by cramertj** [comment]
@alexcrichton Did you want any other changes here?

**Comment by alexcrichton** [comment]
Nah that was it, looks great, thanks!

**Pull request merged**